### DISCUSSION TITLE
dwhci: Transaction failed (status 0x82)

### DISCUSSION THREAD

**Discussion by ArnoldUK** [rsta2/circle#366] [Q&A] [answered]
I'm getting hundreds of these `dwhci: Transaction failed (status 0x82)` messages after a minute or so of booting the kernel. I have a wireless USB keyboard with USB dongle.  I running a keyboard raw capture and decoding test kernel using the `CUSBKeyboardDevice` without any plug and play.

Not seen these messages before and I was curious as to what they mean or what casues them ?

**Comment by rsta2** [comment]
This indicates an USB Transaction Error (e.g. CRC error, bit stuffing error, timeout), which means something went wrong in a transfer on the USB. This is not a serious error and can be ignored normally. You could try to enable the system option `USE_USB_FIQ` for a more accurate timing on the USB.

**Reply by ArnoldUK** [reply]
Cheers for the explaination. I've never seen it before but noticed that it occurs more often when the pi is running alot of code processing in my emulation functions, more often after procesing a raw keyboard handler. I cannot ignore it though as there are hundreds per second and brings the pi to slow crawl (debug messages over serial uart).

I've tried changing `usbspeed=full` and `usbboost=true` in cmdline.txt but seems to make it happen more frequently. I will recompile circle with `USE_USB_FIQ` as I'm using a rpi 1 as I think you also mentioned that this is required for more reliable USB keyboard handling.

EDIT: On a second thought, I'm using the kernel HEX flash method over serial with a USB to serial cable and wondered if this would cause these sort errors ? I only ask this as I've not seen the messages when using the raw kernel copy to the SD-Card and running kernel that way.

Thanks for the super quick support!

**Reply by rsta2** [reply, marked as answer]
If you do not need the FIQ for other purpose and if it helps, `USE_USB_FIQ` is a good solution. I don't think, the bootloader can trigger the USB transaction errors. At the time the USB is initialized in the kernel image, the bootloader has no control at all over the RPi any more.

**Reply by ArnoldUK** [reply]
That appears to have worked now, not a single failed transaction! I must remember to re-compile the circle SDK as I had already defined `USE_USB_FIQ` but missed the important re-compile..doh.

This is awesome, thanks for your help with this.
